packages/salesforcedx-vscode-core/src/settings/sfdxCoreSettings.ts

```diff
@@ -34,6 +35,16 @@ export class SfdxCoreSettings {
     return this.getConfigValue<boolean>(SHOW_CLI_SUCCESS_INFO_MSG, true);
   }
 
+  // checks for Microsoft's telemetry setting as well as Salesforce's telemetry setting.
+  public getTelemetryEnabled(): boolean {
```

Comment:
I check for vscode's telemetry.enabled setting as well as our own telemetry setting.